uber/server.py

```diff
@@ -11,6 +11,25 @@ def _add_email():
 cherrypy.tools.add_email_to_error_page = cherrypy.Tool('after_error_response', _add_email)
 
 
+def on_cherrypy_before_error_response(debug=False):
+    """
+    Write the request headers, and the last error's traceback to the cherrypy error log.
+    Do this all one line so all the information can be collected by external log collectors and easily displayed.
```

Comment:
One extra piece of information we can consider adding but which we might decide we don't want is the page's HTTP parameters.  Right now we'd log GET parameters (by way of logging the request line) but not POST parameters.  That might be a good compromise, since POST parameters might theoretically include someone's password if there's an error on the login page, or something.